vault/cors.go

```diff
@@ -0,0 +1,161 @@
+package vault
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/logical"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+var preflightHeaders = map[string]string{
+	"Access-Control-Allow-Headers":     "*",
+	"Access-Control-Max-Age":           "1800",
```

Comment:
In doing some reading around the web, this is a suggestion to the browser and will not necessarily be honored. Webkit browsers for example will only cache preflight responses for up to 5 minutes, Firefox will apparently hold on to them for up to 24 hours, in which case it would honor values less than that. 
In which case, setting it to 5 minutes would at least be putting webkit browsers and Firefox on the same setting. I wasn't able to find any info about how long IE will hang onto them.